libsel4/tools/cgenerator.py

```diff
@@ -0,0 +1,598 @@
+                CapType("seL4_X86_EPTPT", wordsize),
+                StructType("seL4_VCPUContext", wordsize * 7, wordsize),
+                StructType("seL4_UserContext", wordsize * 12, wordsize),
+            ],
+
+            "x86_64": [
+                Type("seL4_X86_VMAttributes", wordsize, wordsize),
+                CapType("seL4_X86_IOPort", wordsize),
+                CapType("seL4_X86_IOPortControl", wordsize),
+                CapType("seL4_X86_ASIDControl", wordsize),
+                CapType("seL4_X86_ASIDPool", wordsize),
+                CapType("seL4_X86_IOSpace", wordsize),
+                CapType("seL4_X86_Page", wordsize),
+                CapType("seL4_X64_PML4", wordsize),
+                CapType("seL4_X86_PDPT", wordsize),
+                CapType("seL4_X86_PageDirectory", wordsize),
+                CapType("seL4_X86_PageTable", wordsize),
+                CapType("seL4_X86_IOPageTable", wordsize),
+                CapType("seL4_X86_VCPU", wordsize),
+                CapType("seL4_X86_EPTPML4", wordsize),
+                CapType("seL4_X86_EPTPDPT", wordsize),
+                CapType("seL4_X86_EPTPD", wordsize),
+                CapType("seL4_X86_EPTPT", wordsize),
+                StructType("seL4_VCPUContext", wordsize * 7, wordsize),
+                StructType("seL4_UserContext", wordsize * 20, wordsize),
+            ],
+            "riscv32": [
+                Type("seL4_RISCV_VMAttributes", wordsize, wordsize),
+                CapType("seL4_RISCV_Page", wordsize),
+                CapType("seL4_RISCV_PageTable", wordsize),
+                CapType("seL4_RISCV_ASIDControl", wordsize),
+                CapType("seL4_RISCV_ASIDPool", wordsize),
+                StructType("seL4_UserContext", wordsize * 32, wordsize),
+            ],
+            "riscv64": [
+                Type("seL4_RISCV_VMAttributes", wordsize, wordsize),
+                CapType("seL4_RISCV_Page", wordsize),
+                CapType("seL4_RISCV_PageTable", wordsize),
+                CapType("seL4_RISCV_ASIDControl", wordsize),
+                CapType("seL4_RISCV_ASIDPool", wordsize),
+                StructType("seL4_UserContext", wordsize * 32, wordsize),
+            ]
+        }
+        return arch_types
+
+    @staticmethod
+    def generate_param_list(input_params, output_params):
+        # Generate parameters
+        params = []
+        for param in input_params:
+            if not param.type.pass_by_reference():
+                params.append(param.type.render_parameter_name(param.name))
+            else:
+                params.append(param.type.pointer().render_parameter_name(param.name))
+        for param in output_params:
+            if param.type.pass_by_reference():
+                params.append(param.type.pointer().render_parameter_name(param.name))
+
+        return ", ".join(params)
+
+    def _gen_file_header(self):
+        self.contents.append("#ifndef __LIBSEL4_SEL4_CLIENT_H")
```

Comment:
Can we change this to use "`#pragma once`"?

Reply:
Yes, I've removed include guards and used `#pragma once` instead.